TeleCall uses 4 centers around the globe to process customer order forms.They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing. The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences
<br>
Minitab File: CustomerOrderForm.mtw
<br>
Solution:- 
H0 (There is no defective %  varies by centre).
<br>
Ha (There is defective %  varies by centre).

In [30]:
from scipy import stats as st
import pandas as pd
import numpy as np

In [31]:
order = pd.read_csv("Costomer+OrderForm.csv")

In [32]:
CustomerOrder.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [33]:
order.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [34]:
order.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [35]:
order.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [36]:
order.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [56]:
obs = np.array([[271,267,269,280],[29,33,31,20]])

In [64]:
stat, p, dof, exp = st.chi2_contingency(obs)   #o/p --> (Chi2 stats value, p_value, df, expected obsvations)

In [65]:
print('''stat:-{0}\nP_value:-{1}\ndf:-{2}\nexp:-{3}'''.format(stat,p,dof,exp))

stat:-3.858960685820355
P_value:-0.2771020991233135
df:-3
exp:-[[271.75 271.75 271.75 271.75]
 [ 28.25  28.25  28.25  28.25]]


In [66]:
# find Chi-Square critical value
#For 5% significance level(0.05), df = 3
st.chi2.ppf(1-.05, df=3)

7.814727903251179

Here, Statistics = 57.22890784118593, Critical value = 7.814727903251179
<br>
Statistics < 7.82 --> Fail to Reject H0
<br>
Hence, There is no association between centres Also there is no defective % varies by centres.